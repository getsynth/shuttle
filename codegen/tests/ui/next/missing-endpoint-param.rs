shuttle_codegen::app! {
    #[shuttle_codegen::endpoint(method = get)]
    async fn only_method_param() -> &'static str {
        "Hello, World!"
    }

    #[shuttle_codegen::endpoint(route = "/goodbye")]
    async fn only_route_param() -> &'static str {
        "Goodbye, World!"
    }

    #[shuttle_codegen::endpoint()]
    async fn no_params() -> &'static str {
        "Goodbye, World!"
    }
}
