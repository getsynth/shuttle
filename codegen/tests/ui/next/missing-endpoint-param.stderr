error: no route provided

         = help: Add a route to your endpoint: `route = "/hello")`

 --> tests/ui/next/missing-endpoint-param.rs:2:32
  |
2 |     #[shuttle_codegen::endpoint(method = get)]
  |                                ^^^^^^^^^^^^^^

error: no method provided

         = help: Add a method to your endpoint: `method = get`

 --> tests/ui/next/missing-endpoint-param.rs:7:32
  |
7 |     #[shuttle_codegen::endpoint(route = "/goodbye")]
  |                                ^^^^^^^^^^^^^^^^^^^^

error: missing endpoint arguments

         = help: The endpoint takes two arguments: `endpoint(method = get, route = "/hello")`

  --> tests/ui/next/missing-endpoint-param.rs:12:32
   |
12 |     #[shuttle_codegen::endpoint()]
   |                                ^^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/ui/next/missing-endpoint-param.rs:16:2
   |
16 | }
   |  ^ consider adding a `main` function to `$DIR/tests/ui/next/missing-endpoint-param.rs`
